service/featuregate/README.md

```diff
@@ -0,0 +1,111 @@
+# Collector Feature Gates
+
+This package provides a mechanism that allows operators to enable and disable
+experimental or transitional features at deployment time. These flags should
+be able to govern the behavior of the application starting as early as possible
+and should be available to every component such that decisions may be made
+based on flags at the component level.
+
+## Usage
+
+Feature gates must be defined and registered with the global registry in
+an `init()` function.  This makes the `Gate` available to be configured and 
+queried with a default value of its `Enabled` property.
+
+```go
+var fancyNewFeatureGate = featuregate.Gate{
+	ID:          "namespaced.uniqueIdentifier",
+	Description: "A brief description of what the gate controls",
+	Enabled:     false,
```

Comment:
Yes, `false` is equivalent to `Alpha` and `true` to `Beta`.  `Stable` would be covered by removing the gate entirely.  The only value I could see to having an enumeration rather than a bool here is that it would allow leaving `Stable` gates in place and ignoring any attempt to disable them, which could be useful in the case that someone forgets to remove a check for a now-`Stable` gate.